src/hotspot/share/opto/memnode.cpp

```diff
@@ -1403,10 +1403,14 @@ Node* LoadNode::eliminate_autobox(PhaseIterGVN* igvn) {
             // nof_nonstatic_fields so check it anyway
             ciConstant c = box->field_value(ik->nonstatic_field_at(0));
             BasicType bt = c.basic_type();
+
             // Only integer types have boxing cache.
+PRAGMA_DIAG_PUSH
+PRAGMA_FORMAT_OVERFLOW_IGNORED
             assert(bt == T_BOOLEAN || bt == T_CHAR  ||
                    bt == T_BYTE    || bt == T_SHORT ||
                    bt == T_INT     || bt == T_LONG, "wrong type = %s", type2name(bt));
+PRAGMA_DIAG_POP
```

Comment:
The problem here is the definition of type2name, which returns NULL for an unknown BasicType.  It would probably be better if it returned a recognizable string for that situation, eliminating this warning at it's source.

Reply:
While looking at type2name, I noticed the comment for the immediately preceding type2name_tab is wrong.

Reply:
The warning has gone in new commit, and I fixed the comment for `type2name_tab` in it.